pubtools/exodus/_tasks/push.py

```diff
@@ -30,13 +32,21 @@ def run(self):
         publish_id = str(publish.get("id"))
         LOG.info("Publish ID: %s", publish_id)
 
-        for item in self.args.sync:
+        push_items = []
+        with Source.get(self.args.source) as source:
```

Comment:
While I'm not sure this will matter much in practice, maybe not (depends on the size of the pushes), it's generally intended that you should be pushing the items as you find them... not waiting until you find all items.

For instance some larger staging directories themselves can take 30+ minutes just to iterate over, especially if the connection between you and the NFS share isn't super fast.  And there is no reason to wait all that time before starting the first exodus-rsync command, when you could immediately start it as soon as you've got the first item.